enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,227 @@
+Service themselves, something that we did not plan to support but also did not 
+actively reject. We recognize that users consider it now to be an existing feature 
+while we prefer they use APIs and not directly modify managed resources.
+
+The main motivation of this enhancement is to allow the customers to enable 
+and configure `.Spec.LoadBalancerSourceRanges` through a new API.
+
+### User Stories
+
+#### As a cluster admin, when configuring an IngressController, I want to limit access to the load balancer to a specified list of IP address ranges on the cloud provider for this ingress
+
+The cluster admin has the option to modify the IngressController to specify 
+a list of IP address ranges for an IngressController to which access to 
+the load balancer should be restricted. This is available as a Day-2 operation 
+to the cluster admin.
+
+### Goals
+
+Enable the configuration of a range for a load balancer via the `IngressController.Spec`, 
+specifically via the `LoadBalancerStrategy`, with a new parameter `AllowedSourceRanges`, 
+which represents a list of IP address ranges. This is then used by the operator to set
+`.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service in front of the router 
+in cloud environments.
+
+If `.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service is already set by the 
+customer, its value is used to set `AllowedSourceRanges` if and only if `AllowedSourceRanges`
+was not initially set. Otherwise, `.Spec.LoadBalancerSourceRanges` is overwritten by `AllowedSourceRanges`.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+The IngressController API is extended by adding an optional field `AllowedSourceRanges` 
+of type `[]CIDR` to `LoadBalancerStrategy`. If this field is empty, all source addresses 
+are allowed (which is equivalent to specifying "0.0.0.0/0").
+
+```go
+type LoadBalancerStrategy struct {
+	// ...
+
+	// allowedSourceRanges specifies an allowlist of IP address ranges to which
+	// access to the load balancer should be restricted.  Each range must be
+	// specified using CIDR notation (e.g., "10.0.0.0/8").  If no range is
+	// specified, "0.0.0.0/0" is used by default, which allows all source addresses.
+	//
+	// +nullable
+	// +optional
+	AllowedSourceRanges []CIDR `json:"allowedSourceRanges,omitempty"`
+}
+
+// CIDR is an IP address range in CIDR notation (for example, "10.0.0.0/8"
+// or "fd00::/8").
+// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])/([0-9]|[12][0-9]|3[0-2])$`
+type CIDR string
+```
+
+### Workflow Description
+
+1. A cluster admin who wants to limit access to the load balancer to a list of IP address ranges
+of their choice sets the `AllowedSourceRanges` field. 
+2. Ingress Operator reconciles the LoadBalancer Service and sets the `.Spec.LoadBalancerSourceRanges`
+based on `AllowedSourceRanges`.
+
+#### Variation [optional]
+
+N/A
```

Comment:
This looks like a good place to describe the migration workflow.

Reply:
Done.